File: lib/domain/file_blob/dtos/file_blob_purge_output.rb
# frozen_string_literal: true

module Domain
  module FileBlob
    module Dtos
      # 削除要求の結果（対象が存在しなかった場合は `purged` が false）。
      class FileBlobPurgeOutput
        attr_reader :purged

        def initialize(purged:)
          @purged = purged
        end

        def purged?
          purged
        end
      end
    end
  end
end

File: crates/agrr-domain/src/file_blob/dtos/file_blob_purge_output.rs
//! Ruby: `Domain::FileBlob::Dtos::FileBlobPurgeOutput`

/// Ruby: `Domain::FileBlob::Dtos::FileBlobPurgeOutput`
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub struct FileBlobPurgeOutput {
    pub purged: bool,
}

impl FileBlobPurgeOutput {
    pub fn new(purged: bool) -> Self {
        Self { purged }
    }

    pub fn purged(&self) -> bool {
        self.purged
    }
}
